brvs-core/src/main/java/iroha/validation/listener/IrohaReliableChainListener.java

```diff
@@ -0,0 +1,185 @@
+    try {
+      connection = factory.newConnection();
+    } catch (TimeoutException | IOException e) {
+      throw new IllegalStateException("Cannot acquire MQ connection", e);
+    }
+  }
+
+
+  /**
+   * Queries pending transactions for specific users
+   *
+   * @param accountsToMonitor users that transactions should be queried for
+   * @return set of transactions that are in pending state
+   */
+  public Set<Transaction> getAllPendingTransactions(Set<String> accountsToMonitor) {
+    Set<TransactionOuterClass.Transaction> pendingTransactions = new HashSet<>();
+    accountsToMonitor.forEach(account -> {
+          Queries.Query query = Query.builder(account, 1)
+              .getPendingTransactions()
+              .buildSigned(keyPair);
+          pendingTransactions
+              .addAll(irohaAPI.query(query)
+                  .getTransactionsResponse()
+                  .getTransactionsList()
+              );
+        }
+    );
+    return pendingTransactions;
+  }
+
+  /**
+   * Method providing new blocks coming from Iroha by reading MQ
+   *
+   * @return {@link Observable} of Iroha proto {@link QryResponses.BlockQueryResponse} block
+   */
+  public synchronized Observable<BlockOuterClass.Block> getBlockStreaming() {
+    if (!isListening) {
+      executorService.schedule(this::pushIrohaBlocksToMQ, 0, TimeUnit.SECONDS);
+      isListening = true;
+    }
+
+    PublishSubject<Delivery> source = PublishSubject.create();
+    DeliverCallback deliverCallback = (consumerTag, delivery) -> source.onNext(delivery);
+    CancelCallback cancelCallback = consumerTag -> Runnables.doNothing().run();
+    // start consumer
+    try {
+      Channel channel = connection.createChannel();
+      channel.exchangeDeclare(EXCHANGE_RMQ_NAME, BuiltinExchangeType.FANOUT, true);
+      String queue = channel.queueDeclare("", true, false, false, null).getQueue();
+      channel.queueBind(queue, EXCHANGE_RMQ_NAME, "");
+      channel.basicConsume(queue, true, deliverCallback, cancelCallback);
+    } catch (IOException e) {
+      throw new IllegalStateException("Cannot consume Iroha block", e);
+    }
+
+    logger.info("Subscribed to Iroha chain listener");
+
+    return source.map(delivery -> {
+          BlockOuterClass.Block block = iroha.protocol.BlockOuterClass.Block
+              .parseFrom(delivery.getBody());
+          logger.info("Iroha block consumed. Height " + block.getBlockV1().getPayload().getHeight());
+          return block;
+        }
+    );
+  }
+
+  /**
+   * Method pushing new blocks coming from Iroha into MQ
+   */
+  private void pushIrohaBlocksToMQ() {
+    logger.info("Listening Iroha blocks");
+    Queries.BlocksQuery query = new BlocksQueryBuilder(accountId, Instant.now(), 1)
+        .buildSigned(keyPair);
+    Observable<Block> irohaBlocksObservable = irohaAPI.blocksQuery(query)
+        .map(response -> response.getBlockResponse().getBlock());
+
+    Channel channel;
+    try { channel = connection.createChannel();
```

Comment:
This thing is oddly formatted.

Reply:
`Note that closing the channel may be considered good practice, but isn’t strictly necessary here - it will be done automatically anyway when the underlying connection is closed.`